src/components/Search/Search.jsx

```diff
@@ -44,11 +43,15 @@ export const Search = () => {
 
     const handleResultClick = (result) => {
         dispatch(updateLoading(true));
-        dispatch(updateCurrentCoords({ lat: result.lat, lon: result.lon }));
-        dispatch(updateCurrentCity(result.name));
-        fetchWeatherByCoords(result.lat, result.lon)
+        dispatch(
+            updateCurrentSelection({
+                city: result.name,
+                coordinates: { lat: result.lat, lon: result.lat },
+            })
+        );
+        fetchWeatherByCoords(result.lat, result.lat)
```

Comment:
Could you try to use middleware in order to accomplish all these async actions: 1 - update current city selection action, 2 - fetch weather, 3 - dispatch action 1, 4 - dispatch action 2 (hint: this should live in reducer file, you just have to call here for example updateCurrentCity action and all the logic should be executed in the reducer)